### PR TITLE
[checkbox] Add checkbox package for the deprecated android checkbox

### PR DESCRIPTION
**Pull request by byCedric**
# Why

`import { CheckBox } from 'react-native';` is deprecated, this adds it back for Expo SDK and React Native 0.63.

> I'm really not happy with the implementation on Android, it contains [weird bugs like this one](https://github.com/facebook/react-native/blob/v0.63.4/Libraries/Components/CheckBox/CheckBox.android.js#L163). It also is close to not-customizable, which makes it really hard to work within custom apps.

# How

- The checkbox isn't always available, added `isAvailableAsync` check
- Added package by reusing existing react native code, we can add custom code later
- Added bug fixes for color and changed the props to "make it work"
- Added documentation on how to use it

# Test Plan

- Added unit tests to `expo-checkbox` package
- Added checkbox to the native component list

### CODE REVIEW

File: packages/expo-checkbox/unimodule.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "expo-checkbox",
+  "platforms": ["android", "web"]
```
Comment (brentvatne) [inline packages/expo-checkbox/unimodule.json:3]:
```suggestion
  "platforms": ["web"]
```

to avoid autolinking

**Review by brentvatne — APPROVED** [review]

File: packages/expo-checkbox/src/__tests__/ExpoCheckBox-test.android.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import Checkbox from '../Checkbox';
+//
+describe('Checkbox', () => {
+  it('isAvailableAsync resolves to true on android', async () => {
+    await expect(Checkbox.isAvailableAsync()).resolves.toBeTruthy();
+  });
+
+  it('renders a native checkbox', () => {
+    const wrapper = mount(<Checkbox value color="#4630EB" />);
+    // expect(wrapper.find(NativeCheckbox).first()).toBeDefined();
```
Comment (ide) [inline packages/expo-checkbox/src/__tests__/ExpoCheckBox-test.android.tsx:17]: decide what to do with this

File: packages/expo-checkbox/src/__tests__/ExpoCheckBox-test.android.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  });
+
+  it('resolves proper tint colors based on color', () => {
+    const wrapper = mount(<Checkbox color="#4630EB" />);
+    // const nativeProps = wrapper.find(NativeCheckbox).props();
+    // expect(nativeProps).toMatchObject({ tintColors: { true: '#4630EB' } });
```
Comment (ide) [inline packages/expo-checkbox/src/__tests__/ExpoCheckBox-test.android.tsx:23]: decide what to do with these

File: packages/expo-checkbox/src/__tests__/ExpoCheckBox-test.android.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { mount } from 'enzyme';
+import React from 'react';
+// TODO: find out why we get this error on CI, but not locally:
+// "Cannot find module 'react-native/Libraries/Components/Checkbox/Checkbox' from 'ExpoCheckBox-test.android.tsx'""
+// @ts-ignore
+// import NativeCheckbox from 'react-native/Libraries/Components/Checkbox/Checkbox';
```
Comment (ide) [inline packages/expo-checkbox/src/__tests__/ExpoCheckBox-test.android.tsx:6]: make sure this gets looked at soon

**Pull request merged**